Mail/Views/Attachment/AttachmentPreview.swift

```diff
@@ -76,10 +78,58 @@ struct AttachmentPreview: View {
                     }
 
                     Spacer()
+
+                    Button {
+                        guard let attachmentURL = attachment.localUrl else { return }
+                        openInKdrive(with: attachmentURL)
+                    } label: {
+                        Label {
+                            Text(MailResourcesStrings.Localizable.buttonOpenKdrive)
+                                .font(MailTextStyle.labelSecondary.font)
+                        } icon: {
+                            IKIcon(MailResourcesAsset.kdriveLogo, size: .large)
+                        }
+                        .dynamicLabelStyle(sizeClass: sizeClass ?? .regular)
+                    }
                 }
             }
         }
     }
+
+    private func openInKdrive(with url: URL) {
+        guard let destination = writeToGroupContainer(file: url) else { return }
+
+        var targetUrl = URLComponents(string: "kdrive-file-sharing://file")
+        targetUrl?.queryItems = [URLQueryItem(name: "url", value: destination.path)]
+        if let targetAppUrl = targetUrl?.url, urlOpener.canOpen(url: targetAppUrl) {
+            openUrl(targetAppUrl)
+        } else {
+            openUrl(URLConstants.kdriveAppStore.url)
+        }
+    }
+
+    private func writeToGroupContainer(file: URL) -> URL? {
+        guard let sharedContainerURL: URL = FileManager.default
+            .containerURL(forSecurityApplicationGroupIdentifier: Constants.appGroupIdentifier) else { return nil }
+
+        let groupContainer = sharedContainerURL.appendingPathComponent("Library/Caches/file-sharing", conformingTo: .directory)
+        let destination = groupContainer.appendingPathComponent(file.lastPathComponent)
+
+        do {
+            if FileManager.default.fileExists(atPath: groupContainer.path) {
+                try FileManager.default.removeItem(at: groupContainer)
+            }
+            try FileManager.default.createDirectory(at: groupContainer, withIntermediateDirectories: false)
+            try FileManager.default.copyItem(
+                at: file,
+                to: destination
+            )
+            return destination
+        } catch {
+            print("Error copying file: \(error.localizedDescription)")
+            return nil
```

Comment:
Do we really want this method to return nil on error ?
If we move this to `ios-core` I think throwing the error can be helpful for the app that would be using this code.
Also we could add a Sentry and remove the print statement.